### ISSUE TITLE
Filter previous orders by geometry and time

### ISSUE THREAD

**Issue by tw-smith** [stapi-spec/stapi-spec#207]
Currently the spec allows for all previous orders to be returned from the `GET /orders` endpoint. There is a use case to be able to see previously placed orders but only for a specific location and time (e.g. to avoid order duplication). 

This could be done by adding optional `bbox` and `time` fields to the `GET /orders` endpoint to allow a STAC search to be carried out.

https://stacspec.org/en/about/stac-spec/

**Comment by philvarner** [comment]
Yeah, this feels like it should function like the /collections/{collection_id}/items endpoint. Just getting a list of every order you've ever placed in a default ordering (as there's no sort definiton) is not going to be terribly useful.

**Comment by jkeifer** [comment]
Dependent on the answer to #234 to know if we need to support CQL2 for filtering orders here or not.
